rust/src/smb/smb.rs

```diff
@@ -2449,8 +2449,8 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
                 Ok(retval) => { stream_depth = retval as u32; }
                 Err(_) => { SCLogError!("Invalid depth value"); }
             }
-            AppLayerParserSetStreamDepth(IPPROTO_TCP as u8, ALPROTO_SMB, stream_depth);
         }
+        AppLayerParserSetStreamDepth(IPPROTO_TCP as u8, ALPROTO_SMB, stream_depth);
```

Comment:
What happens if we do not set a stream depth?

Reply:
Before https://github.com/OISF/suricata/commit/e5c948df87541509e34445d80c5ef0242da3326f , without setting a stream depth, the default of 0 was used for smb.

After https://github.com/OISF/suricata/commit/e5c948df87541509e34445d80c5ef0242da3326f , without setting a stream depth, the default of 100000 was used for smb as is for other protocols.

This PR resumes the previous behavior

Reply:
Thank you. Then, I changed the behavior of the proto and we should probably have a bug ticket?

Reply:
Done

Do you know if it needs backports ?

Reply:
Doesn't seem like it.